crepecake/src/main/java/com/google/cloud/tools/crepecake/builder/LayerBuilder.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.builder;
+
+import com.google.cloud.tools.crepecake.image.UnwrittenLayer;
+import com.google.cloud.tools.crepecake.tar.TarStreamBuilder;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+
+/** Builds an {@link UnwrittenLayer} from files. */
+public class LayerBuilder {
```

Comment:
This seems like a little light of a wrapper around TarStreamBuilder? There's a lot of late evaluation going on, so I might have this wrong. Is there potential to mimic the `Blobs` model for `Layers`? `Layers.from(TarStreamBuilder xyz)?` or are trying to hide `TarStreamBuilder`?

Reply:
Yeah, this is just trying to hide `TarStreamBuilder` since building a layer should be agnostic to how the archive is built. Rather than having to add files to a layer by constructing a `TarArchiveEntry`, the layer should be constructed with the file to add and the path on the final image filesystem the file belongs to.